task/06-conditions-n-loops-tasks.js

```diff
@@ -332,10 +415,52 @@ function isBracketsBalanced(str) {
  *
  */
 function timespanToHumanString(startDate, endDate) {
-    throw new Error('Not implemented');
+  let res = "";
+  let period = endDate - startDate;
+  if (period <= 45000) {
+    res = "a few seconds ago";
+  } else if (period > 45000 && period <= 90000) {
+    res = "a minute ago";
+  } else if (period > 90000 && period <= 2.7e6) {
+    if (period % 60000 > 30000) {
+      res = Math.ceil(period / 60000) + " minutes ago";
+    } else {
+      res = Math.floor(period / 60000) + " minutes ago";
+    }
+  } else if (period > 2.7e6 && period <= 5.4e6) {
+    res = "an hour ago";
+  } else if (period > 5.4e6 && period <= 7.92e7) {
+    if (period % 3.6e6 > 1.8e6) {
+      res = Math.ceil(period / 3.6e6) + " hours ago";
+    } else {
+      res = Math.floor(period / 3.6e6) + " hours ago";
+    }
+  } else if (period > 7.92e7 && period <= 1.296e8) {
+    res = "a day ago";
+  } else if (period > 1.296e8 && period <= 2.16e9) {
+    if (period % 8.64e7 > 4.32e7) {
+      res = Math.ceil(period / 8.64e7) + " days ago";
+    } else {
+      res = Math.floor(period / 8.64e7) + " days ago";
+    }
+  } else if (period > 2.16e9 && period <= 3.888e9) {
+    res = "a month ago";
+  } else if (period > 3.888e9 && period <= 2.981e10) {
+    if (period % 2.592e9 > 1.296e9) {
+      res = Math.ceil(period / 2.592e9) + " months ago";
+    } else {
+      res = Math.floor(period / 2.592e9) + " months ago";
+    }
+  } else if (period > 2.981e10 && period <= 4.709e10) {
+    res = "a year ago";
+  } else if (period % 3.154e10 > 1.577e10) {
+    res = Math.ceil(period / 3.154e10) + " years ago";
+  } else {
+    res = Math.floor(period / 3.154e10) + " years ago";
+  }
+  return res;
 }
```

Comment:
try to avoid this https://en.wikipedia.org/wiki/Magic_number_(programming)
make time periods variables  in an understandable way